nms/app/packages/magmalte/app/views/network/FEGNetworkDashboard.js

```diff
@@ -0,0 +1,155 @@
+    background: colors.primary.comet,
+    fontFamily: typography.button.fontFamily,
+    fontWeight: typography.button.fontWeight,
+    fontSize: typography.button.fontSize,
+    lineHeight: typography.button.lineHeight,
+    letterSpacing: typography.button.letterSpacing,
+
+    '&:hover': {
+      background: colors.primary.mirage,
+    },
+  },
+}));
+
+/**
+ * Returns the network page of a federation network. It consists of top
+ * bar, which has a button to navigate to the json configuration and a
+ * network information section.
+ */
+export default function NetworkDashboard() {
+  const classes = useStyles();
+  const {history, relativePath, relativeUrl} = useRouter();
+  const ctx = useContext(FEGNetworkContext);
+
+  return (
+    <>
+      <TopBar
+        header="Network"
+        tabs={[
+          {
+            label: ctx?.state?.id || 'Network',
+            to: '/network',
+            icon: NetworkCheck,
+            filters: (
+              <Grid
+                container
+                justify="flex-end"
+                alignItems="center"
+                spacing={2}>
+                <Grid item>
+                  <Button
+                    className={classes.appBarBtn}
+                    onClick={() => {
+                      history.push(relativeUrl('/json'));
+                    }}>
+                    Edit JSON
+                  </Button>
+                </Grid>
+              </Grid>
+            ),
+          },
+        ]}
+      />
+      <Switch>
+        <Route path={relativePath('/json')} component={NetworkJsonConfig} />
+        <Route
+          path={relativePath('/network')}
+          component={NetworkDashboardInternal}
+        />
+        <Redirect to={relativeUrl('/network')} />
+      </Switch>
+    </>
+  );
+}
+
+/**
+ * Returns a json config page which allows a user to edit the network
+ * information.
+ */
+export function NetworkJsonConfig() {
+  const {match} = useRouter();
+  const [error, setError] = useState('');
+  const networkId: string = nullthrows(match.params.networkId);
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const ctx = useContext(FEGNetworkContext);
+
+  return (
+    <JsonEditor
+      content={ctx.state}
+      error={error}
+      onSave={async fegNetwork => {
+        try {
+          ctx.updateNetworks({networkId, fegNetwork});
+          enqueueSnackbar('Network saved successfully', {
+            variant: 'success',
+          });
+          setError('');
+        } catch (e) {
+          setError(e.response?.data?.message ?? e.message);
+        }
+      }}
+    />
+  );
+}
+
+/**
+ * Returns information about the federation network and a table of the servicing
+ * access gateways alongside the serviced networks they are under.
+ */
+export function NetworkDashboardInternal() {
+  const classes = useStyles();
+  const ctx = useContext(FEGNetworkContext);
+
+  return (
+    <div className={classes.dashboardRoot}>
+      <Grid container spacing={4}>
```

Comment:
Do we need these Grid containers ?

Reply:
Oh, I forgot to remove that once I created a network context provider. Since the network context provider now take care of that, I have removed it.

Reply:
Yes, those grid containers are needed for styling purpose. They will make more sense on the next PR I made